cpp/src/Glacier2/SessionRouterI.cpp

```diff
@@ -156,46 +156,56 @@ class UserPasswordCreateSession final : public CreateSession
     }
 
     shared_ptr<FilterManager>
-    createFilterManager() override
+    createFilterManager() final
     {
         return FilterManager::create(_instance, _user, true);
     }
 
     void
-    createSession() override
+    createSession() final
     {
         auto ctx = _current.ctx;
         ctx.insert(_context.begin(), _context.end());
         auto self = shared_from_this();
-        _sessionRouter->_sessionManager->createAsync(_user, _control,
-                                                     [self](SessionPrxPtr response)
-                                                     {
-                                                         self->sessionCreated(std::move(response));
-                                                     },
-                                                     [self](exception_ptr e)
-                                                     {
-                                                         self->createException(e);
-                                                     },
-                                                     nullptr,
-                                                     ctx);
+        _sessionRouter->_sessionManager->createAsync(
+            _user,
+            _control,
+            [self](optional<SessionPrx> session)
+            {
+                if (session)
+                {
+                    self->sessionCreated(std::move(session));
```

Comment:
Why does sessionCreated accept an optional proxy since we know it's always non-null.

Reply:
Actually we also pass nullopt to sessionCreated elsewhere.